token/program-2022/src/lib.rs

```diff
@@ -38,3 +43,9 @@ pub fn check_program_account(spl_token_program_id: &Pubkey) -> ProgramResult {
     }
     Ok(())
 }
+
+/// Checks two pubkeys for equality in a computationally cheap way using
+/// `sol_memcmp`
+pub fn cmp_pubkeys(a: &Pubkey, b: &Pubkey) -> bool {
```

Comment:
Perhaps we could manually implement `PartialEq` for `Pubkey` to make this optimization available to all Rust programs automatically

Reply:
Definitely. I want to wait until the memcmp costs are sorted out and be sure that it's cheaper before making it the default behavior.